aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -848,11 +849,11 @@ ALSO_REGISTER_AVX512_DISPATCH(logit_stub, &CPU_CAPABILITY::logit_kernel);
 ALSO_REGISTER_AVX512_DISPATCH(sinh_stub, &CPU_CAPABILITY::sinh_kernel);
 ALSO_REGISTER_AVX512_DISPATCH(cosh_stub, &CPU_CAPABILITY::cosh_kernel);
 ALSO_REGISTER_AVX512_DISPATCH(atanh_stub, &CPU_CAPABILITY::atanh_kernel);
+ALSO_REGISTER_AVX512_DISPATCH(digamma_stub, &CPU_CAPABILITY::digamma_kernel);
```

Comment:
While it does look like this kernel is compute-intensive & would do better with AVX512 than with AVX2, we can use data to decide whether or not to enable AVX512 for this kernel.
Since your machine does not support AVX512, I'll benchmark with both AVX2 & AVX512, and will share results tomorrow.

 Thanks!

Reply:
Thanks for your patience, @khushi-411!
Please disable AVX512 dispatch for this kernel.
I had also preloaded Intel OpenMP for better performance, and tcmalloc for minimal run-to-run variation.
These measurements were taken on a machine that also supports AVX512_BF16 ISA, which is why BF16 performance was slightly better with AVX512, but we'd address that part in the near future. 

<details>

[------------ torch.digamma(x) Benchmark -----------]
                                        |  vectorized
With AVX512
1 threads: ------------------------------------------
      dtype torch.float16 - n : 100     |      2.1
      dtype torch.float16 - n : 200     |      3.1
      dtype torch.float16 - n : 500     |      6.6
      dtype torch.float16 - n : 1000    |     12.0
      dtype torch.float16 - n : 10000   |    110.4
      dtype torch.float32 - n : 100     |      2.1
      dtype torch.float32 - n : 200     |      3.5
      dtype torch.float32 - n : 500     |      7.4
      dtype torch.float32 - n : 1000    |     14.1
      dtype torch.float32 - n : 10000   |    133.3
      dtype torch.float64 - n : 100     |      2.6
      dtype torch.float64 - n : 200     |      4.5
      dtype torch.float64 - n : 500     |     10.1
      dtype torch.float64 - n : 1000    |     19.3
      dtype torch.float64 - n : 10000   |    185.3
      dtype torch.bfloat16 - n : 100    |      1.9
      dtype torch.bfloat16 - n : 200    |      3.0
      dtype torch.bfloat16 - n : 500    |      6.3
      dtype torch.bfloat16 - n : 1000   |     11.8
      dtype torch.bfloat16 - n : 10000  |    110.9

With AVX2

[------------ torch.digamma(x) Benchmark -----------]
                                        |  vectorized
1 threads: ------------------------------------------
      dtype torch.float16 - n : 100     |      1.9
      dtype torch.float16 - n : 200     |      3.0
      dtype torch.float16 - n : 500     |      6.2
      dtype torch.float16 - n : 1000    |     11.5
      dtype torch.float16 - n : 10000   |    107.1
      dtype torch.float32 - n : 100     |      1.9
      dtype torch.float32 - n : 200     |      3.0
      dtype torch.float32 - n : 500     |      6.3
      dtype torch.float32 - n : 1000    |     11.7
      dtype torch.float32 - n : 10000   |    106.0
      dtype torch.float64 - n : 100     |      2.3
      dtype torch.float64 - n : 200     |      3.9
      dtype torch.float64 - n : 500     |      8.6
      dtype torch.float64 - n : 1000    |     16.3
      dtype torch.float64 - n : 10000   |    155.6
      dtype torch.bfloat16 - n : 100    |      1.9
      dtype torch.bfloat16 - n : 200    |      3.0
      dtype torch.bfloat16 - n : 500    |      6.3
      dtype torch.bfloat16 - n : 1000   |     11.6
      dtype torch.bfloat16 - n : 10000  |    108.5

</details>

Reply:
Thanks for the benchmarks, @sanchitintel! I've updated the PR.

I see float16, float32, and float64 perform better in AVX2 but not in AVX512 and vice versa for bfloat16; just wondering why; Is there any place I could give a read about it.

Reply:
BF16 performs better with AVX2 as well. Sorry, those tables are a bit hard to compare unless copy-pasted side-by-side, just as vfdev-5 did.